apps/system/test/unit/app_chrome_test.js

```diff
@@ -1074,15 +1090,112 @@ suite('system/AppChrome', function() {
       requestStub.restore();
     });
 
-    test('Click pin button', function() {
+    test('Pin page', function() {
+      chrome.pinPage();
+      assert.isTrue(Service.request.calledWith('Places:setPinned',
+        fakeWebSite.url));
+    });
+  });
+
+  suite('Pin site without manifest', function() {
+    var chrome, putStub, resolveUrlStub, getHostnameStub, getIconStub;
+
+    setup(function() {
+      var app = new AppWindow(cloneConfig(fakeWebSite));
+      chrome = new AppChrome(app);
+      resolveUrlStub = this.sinon.stub(UrlHelper, 'resolveUrl').
+        returns('http://google.com/');
+      getHostnameStub = this.sinon.stub(UrlHelper, 'getHostname').
+        returns('google.com');
+      getIconStub = this.sinon.stub(IconsHelper, 'getIcon').
+        returns(Promise.resolve('http://google.com/favicon.ico'));
+      putStub = this.sinon.stub(BookmarksDatabase, 'put').returns(
+        Promise.resolve());
+    });
+
+    teardown(function() {
+      resolveUrlStub.restore();
```

Comment:
I don't think you need to restore the stubs when using this.sinon.stub (only when using the global sinon.stub)